### PR TITLE
feat(tracking): add custom events with distinct id

### PR DESCRIPTION
**Pull request by chohner**
- adds `sendCustomEvent()`
- tries to extract `distinct_id` from posthog cookie, otherwise falls back to `ENVIRONMENT`
- ~⚠️ currently sends ANY submitted step data ⚠️~

TODO
- [x] decide for which steps to fire (and how to name the event)

### CODE REVIEW

File: app/services/analytics/customEvent.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { PostHog } from "posthog-node";
+import { hasTrackingConsent } from "./gdprCookie.server";
+import { config } from "../env/web";
+import { parse } from "cookie";
+
+const { POSTHOG_API_KEY, POSTHOG_API_HOST, ENVIRONMENT } = config();
+const posthogClient = POSTHOG_API_KEY
```
Comment (frederike-ramin) [inline app/services/analytics/customEvent.ts:7]: Should we have multiple instances of this? Feels like one could be enough?

File: app/services/analytics/customEvent.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  ? new PostHog(POSTHOG_API_KEY, { host: POSTHOG_API_HOST })
+  : undefined;
+
+function idFromCookie(request: Request) {
+  // Note: can't use cookie.parse(): https://github.com/remix-run/remix/discussions/5198
+  // Returns ENVIRONMENT if posthog's distinct_id can't be extracted
```
Comment (frederike-ramin) [inline app/services/analytics/customEvent.ts:13]: Just curious - why are we returning ENVIRONMENT and not something like "not given"?

**Review by frederike-ramin — COMMENTED** [review]
Looks fine to me. We should really see if we really want to do this on *every* page submit though.

File: app/services/analytics/customEvent.ts (hunk 1 above)
Comment (chohner) [reply app/services/analytics/customEvent.ts:7]: Great idea to have one implementation for all posthog interactions (although I would hope this init call isn't too expensive)

**Pull request merged**